ignore style-tags

<!--
  Before posting the issue, please confirm that the problem you're getting
  is not related with your code editor configuration.
  To make sure it's not, run: yarn eslint src/your-file.svelte
-->

**Tell us about your environment**

- **ESLint version: 8.57.0
- **@intlify/eslint-plugin-svelte version: 0.4.0
- **Node version: 20.11.1

**Please show your full configuration:**
<!-- Paste content of your .eslintrc file -->

```js
/** @type { import("eslint").Linter.Config } */
module.exports = {
	root: true,
	extends: [
		'eslint:recommended',
		'plugin:@typescript-eslint/recommended',
		'plugin:svelte/recommended',
		'prettier',
		'plugin:@intlify/svelte/recommended',
	],
	parser: '@typescript-eslint/parser',
	plugins: ['@typescript-eslint', '@intlify/svelte', 'unused-imports'],
	rules: {
		// False positives under svelte 5 with generics
		'no-undef': 'off',
		// False positives under svelte 5 with vars named $page etc
		'svelte/valid-compile': 'off',
		'no-unused-vars': 'off',
		'@typescript-eslint/no-unused-vars': [
			'error', // or "error"
			{
				// Variables starting underscores are allowed, becuase it is frequently used
				// Variables starging with $$ are allowed, since shadcn uses it internally (generated code that we edit)
				argsIgnorePattern: '^_',
				varsIgnorePattern: '^(_|\\$\\$)',
				caughtErrorsIgnorePattern: '^_',
			},
		],
		'unused-imports/no-unused-imports': 'error',
		'unused-imports/no-unused-vars': [
			'warn',
			{
				vars: 'all',
				varsIgnorePattern: '^(_|\\$\\$)',
				args: 'after-used',
				argsIgnorePattern: '^_',
			},
		],
	},
	parserOptions: {
		sourceType: 'module',
		ecmaVersion: 2020,
		extraFileExtensions: ['.svelte'],
	},
	env: {
		browser: true,
		es2017: true,
		node: true,
	},
	overrides: [
		{
			files: ['*.svelte'],
			parser: 'svelte-eslint-parser',
			parserOptions: {
				parser: '@typescript-eslint/parser',
			},
		},
	],
}

```


**What did you do?**
<!-- Please include the actual source code causing the issue. -->

1: Created a style-tag with some css in a .svelte-file
2: run eslint on the file ` ./node_modules/.bin/eslint src/routes/test.svelte`

```svelte
<h1>Text</h1>

<style>
	h1 {
		background: blue;
	}
</style>
```


**What did you expect to happen?**

I expected no errors for the style-tag


**What actually happened?**

I got a warning from `@intlify/svelte/no-raw-text` for the css. 

```
./src/routes/test.svelte
  1:5  warning  raw text 'Text' is used                         @intlify/svelte/no-raw-text
  3:8  warning  raw text 'h1 {
		background: blue;
	}' is used  @intlify/svelte/no-raw-text

✖ 2 problems (0 errors, 2 warnings)
```

--- a/.changeset/hungry-news-hug.md
+++ b/.changeset/hungry-news-hug.md
@@ -1,0 +1,5 @@
+---
+"@intlify/eslint-plugin-svelte": patch
+---
+
+fix: false positives for style tag in no-raw-text

--- a/lib/rules/no-raw-text.ts
+++ b/lib/rules/no-raw-text.ts
@@ -185,6 +185,10 @@ function create(context: RuleContext): RuleListener {
       return false
     }
 
+    if (element.type === 'SvelteStyleElement') {
+      return true
+    }
+
     return config.ignoreNodes.includes(
       sourceCode.text.slice(...element.name.range!)
     )
@@ -194,6 +198,7 @@ function create(context: RuleContext): RuleListener {
       | SvAST.SvelteText['parent']
       | SvAST.SvelteMustacheTag['parent']
       | SvAST.SvelteElement
+      | SvAST.SvelteSpecialElement
       | SvAST.SvelteAwaitBlock
       | SvAST.SvelteElseBlockElseIf = node.parent
     while (
@@ -208,7 +213,10 @@ function create(context: RuleContext): RuleListener {
     ) {
       target = target.parent
     }
-    if (target.type === 'SvelteElement') {
+    if (
+      target.type === 'SvelteElement' ||
+      target.type === 'SvelteStyleElement'
+    ) {
       return target
     }
     return null

--- a/tests/lib/rules/no-raw-text.ts
+++ b/tests/lib/rules/no-raw-text.ts
@@ -54,6 +54,14 @@ tester.run('no-raw-text', rule as never, {
         { $_('root level translation') }
       `,
       options: []
+    },
+    {
+      code: `<h1>Text</h1>
+      <style>
+        h1 {
+          background: blue;
+        }
+      </style>`
     }
   ],
 

--- a/tests/lib/rules/no-raw-text.ts
+++ b/tests/lib/rules/no-raw-text.ts
@@ -56,7 +56,7 @@ tester.run('no-raw-text', rule as never, {
       options: []
     },
     {
-      code: `<h1>Text</h1>
+      code: `
       <style>
         h1 {
           background: blue;

